core/session.py

```diff
@@ -217,7 +212,7 @@ def validateSecurityKey(self, key: str) -> bool:
         """
         Checks if key matches the current CSRF-Token of our session. On success, a new key is generated.
         """
-        if compare_digest(self.securityKey, key):
+        if hmac.compare_digest(self.securityKey, key):
```

Comment:
low-hanging-fruit, and more for curiosity: Why should everything a module-import? Only one method from `hmac` is used, so the `from hmac import` is not really longer than writing two times `hmac.` before `compare_digest`. Also, a builtin is not vulnerable for circular import errors.

Please specify such things so that everyone can follow them and the code is consistent.

Reply:
It's more convenient for readability to immediately know that the hmac stdlib function `compare_digest` is used, especially when you don't have the entire context, like in a pull request diff. In the past, I know that such functions where often written by hand. It's more clear to the programmer that standard library features are used here, and its good to use them elsewhere too.